subprojects/composite-builds/src/main/java/org/gradle/composite/internal/CompositeBuildClassPathInitializer.java

```diff
@@ -21,19 +21,33 @@
 import org.gradle.api.internal.initialization.ScriptClassPathInitializer;
 import org.gradle.api.internal.project.ProjectInternal;
 import org.gradle.api.internal.tasks.TaskDependencyUtil;
+import org.gradle.cache.FileLock;
+import org.gradle.cache.FileLockManager;
+import org.gradle.cache.LockOptions;
 import org.gradle.internal.build.BuildState;
+import org.gradle.internal.buildtree.BuildTreeWorkGraph;
 
+import javax.annotation.Nullable;
+import java.io.File;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Set;
 
+import static org.gradle.cache.internal.filelock.LockOptionsBuilder.mode;
+
 public class CompositeBuildClassPathInitializer implements ScriptClassPathInitializer {
```

Comment:
💭 While we are here, could you add a javadoc to this class or to the interface? 🙏